File: clients/java_graal/src/main/java/komus24/model/VehicleOrder.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * TODO - Document
 */
public class VehicleOrder {
    /**
     * -1..+1
     */
    private double accelerate;

    /**
     * -1..+1
     */
    public double getAccelerate() {
        return accelerate;
    }

    /**
     * -1..+1
     */
    public void setAccelerate(double value) {
        this.accelerate = value;
    }
    /**
     * TODO - Document
     */
    private boolean brakes;

    /**
     * TODO - Document
     */
    public boolean isBrakes() {
        return brakes;
    }

    /**
     * TODO - Document
     */
    public void setBrakes(boolean value) {
        this.brakes = value;
    }
    /**
     * -1..+1
     */
    private double rotate;

    /**
     * -1..+1
     */
    public double getRotate() {
        return rotate;
    }

    /**
     * -1..+1
     */
    public void setRotate(double value) {
        this.rotate = value;
    }

    public VehicleOrder(double accelerate, boolean brakes, double rotate) {
        this.accelerate = accelerate;
        this.brakes = brakes;
        this.rotate = rotate;
    }

    /**
     * Read VehicleOrder from input stream
     */
    public static VehicleOrder readFrom(java.io.InputStream stream) throws java.io.IOException {
        double accelerate;
        accelerate = StreamUtil.readDouble(stream);
        boolean brakes;
        brakes = StreamUtil.readBoolean(stream);
        double rotate;
        rotate = StreamUtil.readDouble(stream);
        return new VehicleOrder(accelerate, brakes, rotate);
    }

    /**
     * Write VehicleOrder to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeDouble(stream, accelerate);
        StreamUtil.writeBoolean(stream, brakes);
        StreamUtil.writeDouble(stream, rotate);
    }

    /**
     * Get string representation of VehicleOrder
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("VehicleOrder { ");
        stringBuilder.append("accelerate: ");
        stringBuilder.append(String.valueOf(accelerate));
        stringBuilder.append(", ");
        stringBuilder.append("brakes: ");
        stringBuilder.append(String.valueOf(brakes));
        stringBuilder.append(", ");
        stringBuilder.append("rotate: ");
        stringBuilder.append(String.valueOf(rotate));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/kotlin/src/main/kotlin/komus24/model/VehicleOrder.kt
package komus24.model

import komus24.util.StreamUtil

/**
 * TODO - Document
 */
class VehicleOrder {
    /**
     * -1..+1
     */
    var accelerate: Double
    /**
     * TODO - Document
     */
    var brakes: Boolean
    /**
     * -1..+1
     */
    var rotate: Double

    constructor(accelerate: Double, brakes: Boolean, rotate: Double) {
        this.accelerate = accelerate
        this.brakes = brakes
        this.rotate = rotate
    }

    /**
     * Write VehicleOrder to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeDouble(stream, accelerate)
        StreamUtil.writeBoolean(stream, brakes)
        StreamUtil.writeDouble(stream, rotate)
    }

    /**
     * Get string representation of VehicleOrder
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("VehicleOrder { ")
        stringBuilder.append("accelerate: ")
        stringBuilder.append(accelerate)
        stringBuilder.append(", ")
        stringBuilder.append("brakes: ")
        stringBuilder.append(brakes)
        stringBuilder.append(", ")
        stringBuilder.append("rotate: ")
        stringBuilder.append(rotate)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read VehicleOrder from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): VehicleOrder {
            var accelerate: Double
            accelerate = StreamUtil.readDouble(stream)
            var brakes: Boolean
            brakes = StreamUtil.readBoolean(stream)
            var rotate: Double
            rotate = StreamUtil.readDouble(stream)
            return VehicleOrder(accelerate, brakes, rotate)
        }
    }
}